### ISSUE TITLE
[Bug]: Broken image generation after 1.3.0 merge

### ISSUE THREAD

**Issue by Convexity-ai** [lshqqytiger/stable-diffusion-webui-directml#133]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits

### What happened?

Generating images produces very weird results after the 1.3.0 merge today. All images, including re-running previously known good images, are very blurry and about ¾ of the images are grey.

I’ve tried switching the model, VAE, sampler, turning off all extensions, removing all loras and textual inversion, changing the resolutions, and doing a clean clone today but I can’t seem to get it to work.

The console doesn't show any errors during generation.

See image below for example with revAnimated_v122. Prompt is about a simple as it can get:

```
Prompt: cat
Steps: 40, Sampler: DPM++ 2M Karras, CFG scale: 7, Seed: 1206546347, Size: 640x480, Model hash: 4199bcdd14, Model: revAnimated_v122
```
![00051-1206546347](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/133716788/4e4a4317-43d0-41e5-bf67-17c7f0b6e9cb)

### Steps to reproduce the problem

1. Go to txt2img
2. Type 'cat' into prompt.
3. Set resolution to 640 x 480
4. Set sampler to DPM++ 2M Karras
5. Set seed to 1206546347
6. Press generate

### What should have happened?

Image should be at least somewhat detailed.

### Commit where the problem happens

b4f34ad

### What Python version are you running on ?

Python 3.10.x

### What platforms do you use to access the UI ?

Windows

### What device are you running WebUI on?

AMD GPUs (RX 6000 above)

### What browsers do you use to access the UI ?

Microsoft Edge

### Command Line Arguments

```Shell
--medvram --no-half --no-half-vae --opt-sub-quad-attention --precision full
```

### List of extensions

LDSR | built-in | None |   | Sun May 28 15:53:02 2023 |  
Lora | built-in | None |   | Sun May 28 15:53:02 2023 |  
ScuNET | built-in | None |   | Sun May 28 15:53:02 2023 |  
SwinIR | built-in | None |   | Sun May 28 15:53:02 2023 |  
prompt-bracket-checker | built-in | None |   | Sun May 28 15:53:02 2023

### Console logs

```Shell
Total progress: 100%|██████████████████████████████████████████████████████████████████| 40/40 [00:15<00:00,  2.66it/s]
Total progress: 100%|██████████████████████████████████████████████████████████████████| 40/40 [00:15<00:00,  2.80it/s]
```

**Comment by vnguyen3950** [comment]
This is happening for me as well on a 5700xt.

**Comment by freedessert** [comment]
Having a similar issue after updating and having to resolve via https://github.com/lshqqytiger/stable-diffusion-webui-directml/discussions/131

I have a VAE enabled, Hires fix 2x on, generating random seed. No errors during generation, similar to OP. Only about 5% of the image is generated, the rest is noise. 

>apple on table
Steps: 25, Sampler: DPM++ SDE Karras, CFG scale: 7, Seed: 1553174953, Size: 512x512, Model hash: 3a1e593b6d, Model: nabimix_v1, Denoising strength: 0.55, Hires upscale: 2, Hires steps: 12, Hires upscaler: R-ESRGAN 4x+ Anime6B

![2023-05-28-1553174953-1](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/134864006/ebc417d4-b328-4b2a-868c-693fcff20b1b)

**What Python version are you running on ?**

Python 3.10.x

**What platforms do you use to access the UI ?**

Windows

**What device are you running WebUI on?**

AMD 5700XT

**What browsers do you use to access the UI ?**

Firefox 113.0.2 (64-bit)

**Command Line Arguments**

`--medvram  --opt-sub-quad-attention`

**List of extensions**

![image](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/134864006/fb27d743-940c-4dcb-acd2-97c5733f758a)
 
**Console logs**

``100%|██████████████████████████████████████████████████████████████████████████████████| 25/25 [00:41<00:00,  1.66s/it]
Total prTile 1/962it [05:30,  1.66s/it]
        Tile 2/9
        Tile 3/9
        Tile 4/9
        Tile 5/9
        Tile 6/9
        Tile 7/9
        Tile 8/9
        Tile 9/9
100%|██████████████████████████████████████████████████████████████████████████████████| 12/12 [02:21<00:00, 11.80s/it]
Total progress: 74it [07:59,  6.48s/it]``

**Comment by lshqqytiger** [comment]
I can guess, but I can't work right away due to personal reasons. I will check it tomorrow.
Returning to last commit before merge:
```
$ git checkout 3f52b28025efbc52c2a745d34bc7ed37710c7a1c
```
I'll split `dev` branch from `master`.

**Comment by Dardoth** [comment]
Update: Changing my cross-attention layer optimization from `--opt-sub-quad-attention` to `--opt-split-attention` resolves the issue for me. I would recommend others check this setting.

**Comment by ladydarkness** [comment]
> Update: Changing my cross-attention layer optimization from `--opt-sub-quad-attention` to `--opt-split-attention` resolves the issue for me. I would recommend others check this setting.

It worked for me, also I want thank you [lshqqytiger](https://github.com/lshqqytiger) for your amazing work.

**Comment by songib** [comment]
```
The filename, directory name, or volume label syntax is incorrect.
You are not currently on a branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

venv ".\stable-diffusion-webui-directml\venv\Scripts\Python.exe"
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: <none>
Commit hash: 3f52b28025efbc52c2a745d34bc7ed37710c7a1c
Installing requirements
```

Should I just wait for tomorrow then? IDK how to fix this. 
I still can run it and make some Pictures but a bit worried it may break again. xd

**Comment by mr-september** [comment]
> Update: Changing my cross-attention layer optimization from `--opt-sub-quad-attention` to `--opt-split-attention` resolves the issue for me. I would recommend others check this setting.

This has worked for me as well.

I'm curious, after @lshqqytiger fixes this issue, should we go back to `--opt-sub-quad-attention`? Or is staying with `--opt-split-attention` better?

**Comment by MaVe-64** [comment]
> Should I just wait for tomorrow then? IDK how to fix this. I still can run it and make some Pictures but a bit worried it may break again. xd
> 
> Something like this? `git pull --set-upstream https://github.com/lshqqytiger/stable-diffusion-webui-directml master`

Just load last commit by typing:
git checkout 3f52b28025efbc52c2a745d34bc7ed37710c7a1c

**Comment by morynet** [comment]
and working

**Comment by Dardoth** [comment]
In testing, although `--opt-split-attention` did resolve the issue above, I went back to the 1.2.1 merge as `--opt-sub-quad-attention` is a lot better at memory management and I'm able to generate much larger images with that setting.

**Comment by Dardoth** [comment]
> It seems to work properly now. Pull [5e48f30](https://github.com/lshqqytiger/stable-diffusion-webui-directml/commit/5e48f300c1b74f0312f692dc8d0485b5120fc497) and try again.

Pulled the latest build and tried again. Image generation with `--opt-sub-quad-attention` seems to work properly now. Thanks for the fix.

**Comment by LappIand** [comment]
I am still having an issue with image generation. I tried the command line args, they weren't yielding a different result... Could somebody explain to me how they got it to work on AMD? Thank you !
![00002-2991867549](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/68433453/da6a4cef-ee2a-44d3-b08e-ddcf311e0d2a)

**Comment by tonykero** [comment]
@LappIand 
I had strange artefacts altough different from what got posted here, but this fixed it for now.


> Just load last commit by typing: `git checkout 3f52b28025efbc52c2a745d34bc7ed37710c7a1c`

**Comment by sgtsixpack** [comment]
> In testing, although `--opt-split-attention` did resolve the issue above, I went back to the 1.2.1 merge as `--opt-sub-quad-attention` is a lot better at memory management and I'm able to generate much larger images with that setting.

This is what I found too.  Also thanks for the commit reversion.  I didn't even need to type:
`git checkout 3f52b28025efbc52c2a745d34bc7ed37710c7a1c`

**Comment by paulKra** [comment]
I appreciate the quick fix, but are we sure that there still isn't a regression, I for example still can't generate(i get an OOM in  line 213, in efficient_dot_product_attention) a 680x512 with 2 Controlnet models active, which i could before, and still i can if i checkout a older version(108ada8) of sub_quadratic_attention.py, 

```
 # TODO: maybe we should use torch.empty_like(query) to allocate storage in-advance,
-    # and pass slices to be mutated, instead of torch.cat()ing the returned slices
````

This is from a diff, is it possible that allocating in advance might have sth to do with it? Or is the problen entirely on my end?

**Comment by LetsLuilu** [comment]
Still get issue with image generation. I reinstalled it, tried different command arg and I still get these bad generations.
I have a amd.

**Comment by lshqqytiger** [comment]
> Still having an issue with image generation. I reinstalled it, tried different command arg and I still get these bad generations.
I have a amd.

Why incoherent images are generated is very diverse.
More informations (example image) are needed.

**Comment by NoAvailableAlias** [comment]
IIRC I thought it was embeddings causing this anomalous output but then I found #142 and adding --no-half to command args resolved it.

**Comment by lshqqytiger** [comment]
> > > Still having an issue with image generation. I reinstalled it, tried different command arg and I still get these bad generations.
> > > I have a amd.
> > 
> > Why incoherent images are generated is very diverse. More informations (e.g. generated incoherent image) are needed.
> 
> ![tmpfelt_y9v](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/135063249/04975d85-2847-48ef-817e-9bc87677c7cd)
> I only get pictures like this. Sometimes he also generates what I want but extremely poorly and it also happens only rarely.

This is not the case of this issue.
What commandline arguments did you use?

**Comment by LetsLuilu** [comment]
> > > > Still having an issue with image generation. I reinstalled it, tried different command arg and I still get these bad generations.
> > > > I have a amd.
> > > 
> > > 
> > > Why incoherent images are generated is very diverse. More informations (e.g. generated incoherent image) are needed.
> > 
> > 
> > ![tmpfelt_y9v](https://user-images.githubusercontent.com/135063249/242354024-04975d85-2847-48ef-817e-9bc87677c7cd.png)
> > I only get pictures like this. Sometimes he also generates what I want but extremely poorly and it also happens only rarely.
> 
> This is not the case of this issue. What commandline arguments did you use?

i use --opt-sub-quad-attention --lowvram --disable-nan-check --autolaunch. I noticed that it may be due to the embeddings I have from huggingface. Whenever I generate with embeddings/negative prompts  then the images become very bad, without they become better but still not as good as before the update.

**Comment by lshqqytiger** [comment]
Try again with `--no-half`. Without that, it can generate incoherent images as you reported. (It's random)

**Comment by Miraihi** [comment]
Okay. In the new version there is now a brand new settings menu: "Optimizations". So delete your optimization arguments from webui-user. sub-quad-attention is especially broken right now.
[Here are my settings that work for me pretty well](https://i.imgur.com/1Z5bdog.png).

**Comment by FrakerKill** [comment]
> Okay. In the new version there is now a brand new settings menu: "Optimizations". So delete your "opt" optimization arguments from webui-user. sub-quad-attention is especially broken right now. [Here are my settings that seem to work pretty well](https://i.imgur.com/1Z5bdog.png).

And in webui-user what commands do you have?

**Comment by sgtsixpack** [comment]
> Okay. In the new version there is now a brand new settings menu: "Optimizations". So delete your "opt" optimization arguments from webui-user. sub-quad-attention is especially broken right now. [Here are my settings that seem to work pretty well](https://i.imgur.com/1Z5bdog.png).

I just tried your settings and it takes 4min per 1024 generation whereas before it was like 1min.

**Comment by Miraihi** [comment]
> I just tried your settings and it takes 4min per 1024 generation whereas before it was like 1min.

The optimal arguments depend on your graphics card. Find out what's working for you. My point was that there's now a dedicated option of `opt` variant inside the web interface, so you don't need one in the arguments.

**Comment by mr-september** [comment]
> Okay. In the new version there is now a brand new settings menu: "Optimizations". So delete your "opt" optimization arguments from webui-user. sub-quad-attention is especially broken right now. [Here are my settings that seem to work pretty well](https://i.imgur.com/1Z5bdog.png).

I reverted to sub-quad-attention and surprisingly haven't run into any issues yet. Not sure if it's been patched or it's specific to GPUs?

> I just tried your settings and it takes 4min per 1024 generation whereas before it was like 1min.

What was your settings? And GPU?

**Comment by Miraihi** [comment]
> I reverted to sub-quad-attention and surprisingly haven't run into any issues yet. Not sure if it's been patched or it's specific to GPUs?

The problem arises only when you try to activate the token merging (Breaks down at 0.1 already).

**Comment by sgtsixpack** [comment]
My settings:
--opt-sub-quad-attention --autolaunch

x6800xt.  I actually need that setting or I was getting problems without it.
